akka-http-core/src/main/scala/akka/http/impl/util/package.scala

```diff
@@ -63,6 +63,13 @@ package object util {
         case Failure(ex) ⇒ throw new RuntimeException("Trying to await result of failed Future, see the cause for the original problem.", ex)
       }
     }
+    def awaitWithException(atMost: Duration): T = {
```

Comment:
`awaitResult` also keeps the exception as the cause but does also include an exception to find out at which place it was called. So, no need to introduce another method.